week02/ValidParentheses.cpp

```diff
@@ -0,0 +1,29 @@
+// TC: O(string_size)
+// MC: O(string_size)
+
+class Solution {
+public:
+    bool isValid(string s) {
+
+        stack <char> opennings;
+        bool valid = 1;
+        for(int i = 0; i < s.size(); i++){
+            if(s[i] == '(' || s[i] == '{' || s[i] == '[')
+                opennings.push(s[i]);
+            else{
+                if(!opennings.empty()){
+                    bool found = 0;
+                    if(opennings.top() == '(' && s[i] == ')') found = 1;
+                    if(opennings.top() == '{' && s[i] == '}') found = 1;
+                    if(opennings.top() == '[' && s[i] == ']') found = 1;
```

Comment:
What if I add/remove another kind of parentheses? You need to add separate if condition for that too